src/ruleset.rs

```diff
@@ -653,6 +653,32 @@ impl RulesetCreated {
         };
         Ok(body()?)
     }
+
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set. The `RulesetCreated` object that is produced is intended
+    /// to be a readonly copy of the original object, however this
+    /// is not currently supported by the landlock kernel module. The intention
+    /// behind this method is to provide a way to keep a parent ruleset intact so
+    /// it can be applied to any number of new processes or threads without
+    /// creating the object over and over.
+    ///
+    /// Any modification of the underlying ruleset changes will alter both
+    /// the new RulesetCreated and the original.
+    ///
+    /// On error, returns a wrapped [`RulesetError::NewRef`].
+    pub fn new_ref(&self) -> Result<RulesetCreated, std::io::Error> {
+        let dup_fd = unsafe { libc::fcntl(self.fd, libc::F_DUPFD_CLOEXEC, 0) };
```

Comment:
I changed to `try_clone`. I think changing to `OwnedFd` is a good idea, but it will require more work to make the rest of the compat state work correctly since `OwndFd` can't be an invalid descriptor which is what happens now with a few states. This is out of scope of this particular work item (for me at least).

Reply:
OK, I'll work on this `OwnedFd` change after your PR.